### PR TITLE
Add URL swapping for marketplace based on environment

### LINKED ISSUE #8371: [Market] Allow use of dev marketplace in dev project
Currently, we can use only the local marketplace or the production market place. We should be able to swap the URL based on dev or prod environment in marketplace

### PR DESCRIPTION
**Pull request by Abhi1992002**
### Fixes #8371

These changes are needed to automatically switch between local and production marketplace URLs, ensuring the app connects to the correct environment (dev or prod) without manual intervention.

### Changes 🏗️

1. Swaps marketplace URL based on APP_ENV (dev or prod).
2. Ensures correct URL is used for local or production environments.

### CODE REVIEW

File: autogpt_platform/frontend/src/components/nav/MarketPopup.tsx [hunk 1]
```diff
@@ -7,7 +7,9 @@ interface MarketPopupProps extends ButtonHTMLAttributes<HTMLButtonElement> {
 
 export default function MarketPopup({
   className = "",
-  marketplaceUrl = "http://platform.agpt.co/marketplace",
+  marketplaceUrl = process.env.APP_ENV === "prod"
+    ? "https://production-marketplace-url.com"
```
Comment (ntindle) [inline autogpt_platform/frontend/src/components/nav/MarketPopup.tsx]: can we keep the existing link for prod and add  a spot for one for "dev": https://dev-builder.agpt.co/marketplace and finally have the localhost fallback for local

**Review by ntindle — CHANGES_REQUESTED** [review]

**Review by ntindle — APPROVED** [review]

File: autogpt_platform/frontend/src/components/nav/MarketPopup.tsx [hunk 2]
```diff
@@ -7,7 +7,15 @@ interface MarketPopupProps extends ButtonHTMLAttributes<HTMLButtonElement> {
...
-  marketplaceUrl = "http://platform.agpt.co/marketplace",
+  marketplaceUrl = (() => {
+    if (process.env.APP_ENV === "prod") {
+      return "https://production-marketplace-url.com";
+    } else if (process.env.APP_ENV === "dev") {
+      return "https://dev-builder.agpt.co/marketplace";
+    } else {
+      return "http://localhost:3000/marketplace";
+    }
+  })(),
```
Comment (ntindle) [inline autogpt_platform/frontend/src/components/nav/MarketPopup.tsx]: This isn't actually accessible on the frontend in testing :/ i think we will need a different way to handle this

**Review by ntindle — CHANGES_REQUESTED** [review]

File: autogpt_platform/frontend/src/components/nav/MarketPopup.tsx (hunk 2 above)
Comment (aarushik93) [reply autogpt_platform/frontend/src/components/nav/MarketPopup.tsx]: I think you want to use the build variables

File: autogpt_platform/frontend/src/components/nav/MarketPopup.tsx [hunk 3]
```diff
@@ -7,7 +7,15 @@ interface MarketPopupProps extends ButtonHTMLAttributes<HTMLButtonElement> {
 
 export default function MarketPopup({
   className = "",
-  marketplaceUrl = "http://platform.agpt.co/marketplace",
+  marketplaceUrl = (() => {
+    if (process.env.NEXT_PUBLIC_APP_ENV === "prod") {
+      return "https://production-marketplace-url.com";
+    } else if (process.env.NEXT_PUBLIC_APP_ENV === "dev") {
```
Comment (aarushik93) [inline autogpt_platform/frontend/src/components/nav/MarketPopup.tsx:13]: this is a new variable, right, can you please add it to the .env example files in frontend

**Pull request merged**